test/simulation/install-geth.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+set -e
+GETH_SOURCE=$1
+GETH_PATH=$2
+EXPECTED_HASH=$3
+install_geth () {
+  echo "installing geth"
+  git clone --depth 1 "$GETH_SOURCE" "$GETH_PATH"
+  cd "$GETH_PATH" || exit 1
+  GOPATH=$GO_PATH CI=false GO111MODULE=off go run build/ci.go install ./cmd/geth
+}
+if [ ! -d "$GETH_PATH" ]; then
+  install_geth
+else
+  ROOT=$PWD
+  cd "$GETH_PATH" || exit 1
```

Comment:
What does the `|| exit 1` do? Exit the script with the code 1 if `cd "$GETH_PATH"` fails?

Reply:
Exactly. https://github.com/koalaman/shellcheck/wiki/SC2164

Since we're using `set -e` at the beginning of the script and will immediately exit the script with a non-zero status upon command failure it doesn't make sense to have these. I removed the redundant code.